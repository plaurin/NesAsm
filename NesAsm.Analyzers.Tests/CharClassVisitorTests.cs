using ApprovalTests;
using ApprovalTests.Reporters;
using NesAsm.Analyzers.Visitors;

namespace NesAsm.Analyzers.Tests;

[UseReporter(typeof(VisualStudioReporter))]
public class CharClassVisitorTests
{
    [Fact]
    public void Test()
    {
        var result = CharClassVisitor.Process(@"C:\Users\pasca\Dev\GitHub\NesAsm\NesAsm.Example\game1.png");

        Approvals.Verify(result);
    }

    [Fact]
    public void TestImportCharScript()
    {
        TestScript("ImportCharDefinition.cs");
    }

    private static void TestScript(params string[] filenames)
    {
        var generator = new CharGenerator();

        var sourcePaths = filenames.Select(filename => $@"..\..\..\TestFiles\{filename}").ToArray();

        GeneratorTestUtilities.TestGenerator(generator, sourcePaths);
    }
}