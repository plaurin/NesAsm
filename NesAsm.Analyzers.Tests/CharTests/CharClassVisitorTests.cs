using ApprovalTests.Reporters;

namespace NesAsm.Analyzers.Tests.CharTests;

[UseReporter(typeof(VisualStudioReporter))]
public class CharClassVisitorTests
{
    [Fact]
    public void TestImportCharScript()
    {
        TestScript("ImportCharDefinition.cs");
    }

    [Fact]
    public void TestInvalidSizeCharScript()
    {
        TestScript("InvalidSize.cs");
    }

    [Fact]
    public void TestMoreThanFourPalettesCharScript()
    {
        TestScript("MoreThanFourPalettes.cs");
    }

    [Fact]
    public void TestMoreThanThreeNonDefaultColorCharScript()
    {
        TestScript("MoreThanThreeNonDefaultColor.cs");
    }

    [Fact]
    public void TestColorMismatchCharScript()
    {
        TestScript("ColorMismatch.cs");
    }

    private static void TestScript(params string[] filenames)
    {
        var generator = new CharGenerator();

        var sourcePaths = filenames.Select(filename => $@"..\..\..\CharTests\TestFiles\{filename}").ToArray();

        GeneratorTestUtilities.TestGenerator(generator, sourcePaths);
    }
}