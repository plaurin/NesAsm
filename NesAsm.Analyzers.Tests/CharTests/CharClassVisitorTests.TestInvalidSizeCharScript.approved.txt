- Main diagnostics:
..\..\..\CharTests\TestFiles\InvalidSize.cs(6,2): error NA1001: Png image size should be 143 by 287 but was 18 by 18
- SyntaxTrees Count: 3
- Output diagnostics:
- RunResult.GeneratedTrees Count: 2
- RunResults diagnostics:
..\..\..\CharTests\TestFiles\InvalidSize.cs(6,2): error NA1001: Png image size should be 143 by 287 but was 18 by 18
- RunResult[0].Generator: Microsoft.CodeAnalysis.IncrementalGeneratorWrapper
- RunResult[0] diagnostics:
..\..\..\CharTests\TestFiles\InvalidSize.cs(6,2): error NA1001: Png image size should be 143 by 287 but was 18 by 18
- RunResult[0].GeneratedSources Count: 2
- RunResult[0].Exception: 
- RunResult[0].GeneratedSources[0].HintName: Asm.InvalidSize.cs
- RunResult[0].GeneratedSources[0].SourceText:
/* Generator Asm code in file Asm.InvalidSize.s
; Auto generated code using the NesAsm project

*/
- RunResult[0].GeneratedSources[1].HintName: Sharp.InvalidSize.cs
- RunResult[0].GeneratedSources[1].SourceText:
// Auto generated code using the NesAsm project

